reference/replica-rebuild.md

```diff
@@ -1,10 +1,82 @@
+-->
+
+**Full rebuild** and **partial rebuild** are two approaches used in the context of replica redundancy and data recovery in a distributed storage system.
+
+### Full rebuild
+
+A full rebuild refers to the process of completely reconstructing a faulty replica of a volume target in a distributed storage system. In this process, when the control plane detects a replica has faulted, possibly due to node failure, connectivity or other issues (**degraded state**), it removes the faulty replica and initiates the creation of a new replica. The new replica is then added to the volume. Once the new replica is added, the data-plane internally triggers a full rebuild by copying entire data from an already existing healthy replica into the newly added one.
+
+:::note
+A drawback of the full rebuild approach is that even if a replica becomes temporarily inaccessible due to reasons like a node restart, a full rebuild is triggered. This means that the entire replica, regardless of its size, is reconstructed from scratch. This can be time-consuming and inefficient, especially for large replicas.
+:::
+
+To address the drawback of full rebuild issue, a partial rebuild feature is introduced. 
+
+### Partial rebuild
+
+In the partial rebuild approach, when a volume target encounters an I/O error on a child/replica, it marks that child as **faulted**, effectively removing it from the I/O path. Simultaneously, the volume target starts maintaining a write log for all subsequent writes.
+
+The Core agent, responsible for managing the replica status, waits for a pre-configured period, known as the **faulted child wait period**. During this time, it expects the replica to come back online. If the faulted replica becomes online within the timeout, the control plane instructs the volume target to bring the child online and include it in the I/O path. Additionally, a partial rebuild process is initiated using the previously recorded write log. The partial rebuild focuses only on the changes made since the replica was marked as faulty, thereby reducing the time and resources required compared to a full rebuild.
+
+By implementing a partial rebuild strategy, the storage system can achieve faster rebuild times and avoid unnecessary full rebuilds for replicas that experience temporary interruptions or failures.
 
 {% hint style="info %}
 The control plane waits for 10 minutes before initiating the full rebuild process, as the `--faulted-child-wait-period` is set to 10 minutes. To configure this parameter, edit values.yaml.
-{% endhint %}
\ No newline at end of file
+{% endhint %}
+
+## Replica rebuild history 
+
+The data-plane handles both full and partial replica rebuilds. To view the logs of these rebuilds, you can use either the `kubectl` command or make an API call.
```

Comment:
`To view history of the rebuilds that an existing volume target has undergone during its lifecycle till now...`